### PR TITLE
cargo fmt + clippy

### PR DESCRIPTION
**Pull request by scabrero**
Run cargo fmt and fix clippy errors

Checklist

- [ ] This pr contains no AI generated code
- [X] cargo fmt has been run
- [X] cargo clippy has been run
- [X] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
Hey @scabrero can you check what version of rust/cargo you are using? We normally use the latest, and sometimes cargo fmt has minor differences between versions. We normally cargo fmt the whole code base during every PR, and so it's surprising to see so many changes.

You should be on cargo/rust 1.73 - if you're not, likely you need to update :)

File: server/lib/src/idm/authsession.rs [hunk 1]
```diff
@@ -714,6 +714,15 @@ impl AuthSessionState {
     }
 }
 
+pub(crate) struct AuthSessionData<'a> {
```
Comment (yaleman) [inline server/lib/src/idm/authsession.rs:717]: This isn't just a clippy/fmt change and should probably be out in `kanidm_proto`

**Review by yaleman — CHANGES_REQUESTED** [review]

**Comment by Firstyear** [comment]
> > Hey @scabrero can you check what version of rust/cargo you are using? We normally use the latest, and sometimes cargo fmt has minor differences between versions. We normally cargo fmt the whole code base during every PR, and so it's surprising to see so many changes.
> > You should be on cargo/rust 1.73 - if you're not, likely you need to update :)
> 
> Ah, that fixed the cargo fmt differences, but brought a new set of clippy errors :) I will update this PR fixing these new ones.

Some of the clippy errors are around unwraps that I intentionally left in as I'm slowly working to bubble up some errors to higher levels. So if you want to take that on, go for it, but if you see "unwrap() shouldn't be used" feel free to leave that for me to clean up in the coming days :)

**Comment by yaleman** [comment]
This needs another rebase but otherwise looks good to go 👍🏻

File: server/lib/src/idm/authsession.rs (hunk 1 above)
Comment (Firstyear) [reply server/lib/src/idm/authsession.rs:717]: This would be an internal struct too, not proto.

**Review by Firstyear — APPROVED** [review]

**Review by yaleman — APPROVED** [review]

**Pull request merged**